### PR TITLE
[Defapp] Use real HPCON for PTY management; Have Monarch always listen for connections

### LINKED ISSUE #9464: [defapp] Clean up hpcon
TODO:
- [ ] - TODO hpcon in ConPtyConnection

_Originally posted by @miniksa in https://github.com/microsoft/terminal/pull/7489#issuecomment-789230966_

### PR DESCRIPTION
**Pull request by miniksa**
[Defapp] Use real HPCON for PTY management; Have Monarch always listen for connections

## PR Checklist
* [x] Closes #9464
* [x] Related to #9475 - incomplete fix
* [x] I work here.
* [x] Manual test

## Detailed Description of the Pull Request / Additional comments
- Sometimes peasants can't manage to accept a connection appropriately because I wrote defterm before @zadjii-msft's monarch/peasant architecture. The simple solution here is to just make the monarch always be listening for inbound connections. Then COM won't start a peasant with -Embedding just to ask the monarch where it should go. It'll just join the active window. I didn't close 9475 because it should follow monarch policies on which window to join... and it doesn't yet.
- A lot of interesting things are happening because this didn't have a real HPCON. So I passed through the remaining handles (and re-GUID-ed the interface) that made it possible for me to pack the right process handles and such into an HPCON on the inbound connection and monitor that like any other ConptyConnection. This should resolve some of the process exit behaviors and signal channel things like resizing.

### CODE REVIEW

**Comment by DHowett** [comment]
How did the bot miss Psuedo

**Comment by miniksa** [comment]
> How did the bot miss Psuedo

psuedoconsole was in `expect.txt`. I've removed it and will see what bot says about this.

File: src/cascadia/TerminalConnection/CTerminalHandoff.cpp [hunk 1]
```diff
@@ -101,8 +103,10 @@ HRESULT CTerminalHandoff::EstablishPtyHandoff(HANDLE in, HANDLE out, HANDLE sign
     RETURN_IF_FAILED(_duplicateHandle(in, in));
     RETURN_IF_FAILED(_duplicateHandle(out, out));
     RETURN_IF_FAILED(_duplicateHandle(signal, signal));
-    RETURN_IF_FAILED(_duplicateHandle(process, process));
+    RETURN_IF_FAILED(_duplicateHandle(ref, ref));
+    RETURN_IF_FAILED(_duplicateHandle(server, server));
+    RETURN_IF_FAILED(_duplicateHandle(client, client));
```
Comment (DHowett) [inline src/cascadia/TerminalConnection/CTerminalHandoff.cpp:108]: It seems like we are duplicating the handles *twice*. Is that so? Once here and once in PackPseuedo

**Review by DHowett — APPROVED** [review]

File: src/cascadia/TerminalConnection/CTerminalHandoff.cpp (hunk 1 above)
Comment (miniksa) [reply src/cascadia/TerminalConnection/CTerminalHandoff.cpp:108]: Yeah it's probably too many times but I haven't fully shaken that out yet. Do you want me to pull one and risk it or just ride with it for now?

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 2]
```diff
@@ -1664,7 +1664,10 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     void TermControl::_CursorPositionChanged(const IInspectable& /*sender*/,
                                              const IInspectable& /*args*/)
     {
-        _tsfTryRedrawCanvas->Run();
+        if (_tsfTryRedrawCanvas)
```
Comment (zadjii-msft) [inline src/cascadia/TerminalControl/TermControl.cpp:1667]: this seems sus - why is this in this PR? I don't doubt that this makes sense, but I'm just wondering. I suppose this is possible to hit if we've started closing, and we've `.reset()` `_tsfTryRedrawCanvas` in `Close`, but we're not fully dtor'd yet.

for the record - leave this in. Just curious to see it here.

Comment (miniksa) [reply src/cascadia/TerminalControl/TermControl.cpp:1667]: It was crashing on me while I was doing things so I guarded it.

File: src/cascadia/TerminalConnection/CTerminalHandoff.cpp (hunk 1 above)
Comment (miniksa) [reply src/cascadia/TerminalConnection/CTerminalHandoff.cpp:108]: Oh hm I guess I have leftovers if I do it this way. OK. I'll remove it from pack.

**Review by zadjii-msft — APPROVED** [review]
I'd love to have more info on what difference duplicating the handles twice does, but the rest of this looks fine. Maybe spin one build with both duplications and one without, and see what the difference has. I don't want to leak processes when we're closed, but I also don't want to crash accidentally.

**Comment by miniksa** [comment]
> I'd love to have more info on what difference duplicating the handles twice does, but the rest of this looks fine. Maybe spin one build with both duplications and one without, and see what the difference has. I don't want to leak processes when we're closed, but I also don't want to crash accidentally.

Dual duplication is a leak because I don't free one of the copies. I wanted the pty-api-like func to not trust the caller so it duped. The other dupe was because it HAS to on the way in the COM call. But my func signature isn't wil::unique_handle so nothing tracks ownership, the bare HANDLE is just a number. So anyone closing it ruins it for the whole app.

However, I realize I wrote the Pack API quickly for us only so I can refine that later if necessary and ONE dupe overall in this process space (that is tracked... in this case by holding the HPCON) is sufficient.

**Comment by DHowett** [comment]
@msftbot merge this in 1 minute

**Pull request merged**